src/mainboard/raptor-cs/talos-2/Kconfig

```diff
@@ -27,6 +27,23 @@ config BOARD_SPECIFIC_OPTIONS
 	select IPMI_BT_ROMSTAGE
 	select FLATTENED_DEVICE_TREE
 
+config TALOS_2_INFINEON_TPM_1
```

Comment:
This can't be define like that. MAINBOARD_HAS_TPM1 should be selected by BOARD_SPECIFIC_OPTIONS and then TPM_MEASURED_BOOT and MAINBOARD_HAS_TPM1 should be selectable in the menuconfig or saved in miniconfig file.

Reply:
> MAINBOARD_HAS_TPM1 should be selected by BOARD_SPECIFIC_OPTIONS and then TPM_MEASURED_BOOT and MAINBOARD_HAS_TPM1 should be selectable in the menuconfig or saved in miniconfig file.

I thought that including it in `BOARD_SPECIFIC_OPTIONS` will enable it unconditionally, no? Or do you have a typo somewhere? I didn't include `MAINBOARD_HAS_TPM1` in `BOARD_SPECIFIC_OPTIONS` because mainboard can work with TPM2 as well, or should it be selected conditionally if `MAINBOARD_HAS_TPM1` is on?

Reply:
This may be undiscovered territory, most (all?) of I2C TPMs used by coreboot are probably soldered. AFAICT only `mainboard/google/gru` has support for different TPM families. I suggest something similar to that.